hw/mcu/dialog/da1469x/src/da1469x_snc.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <assert.h>
+#include "mcu/mcu.h"
+#include "mcu/da1469x_snc.h"
+
+int
+snc_sw_init(void)
```

Comment:
this should acquire PD_PER power domain (`da1469x_pd_acquire(MCU_PD_DOMAIN_COM)`) as otherwise it may be turned off when going to deep sleep

Reply:
Yep; it definitely should.

Reply:
Actually, I have a comment about this one. So, there is a signal from the SNC called SNC_DONE that goes to the PDC. In the case where the device is SW controlled that signal will still come from the SNC but you will still have to manually release the COM domain with da1469x_pd_release(). Since there is no specific API to release the SNC from SW control it seems a bit odd to have the SNC code acquire but not release the COM domain. So I think we have two choices here:
1) The user must separately call pd_acquire and pd_release.
2) Add an API to the SNC code that releases the SNC from SW control. This would call pd_release.

I am leaning towards 2). Do you have a preference?

Reply:
I prefer 2 and the reason is that PDC code resides in MCU package and should not be really accessed from application, only from MCU/BSP code and drivers.